service/src/main/java/bio/terra/cbas/dependencies/wds/WdsService.java

```diff
@@ -57,6 +83,131 @@ public RecordResponse updateRecord(
         });
   }
 
+  public Capabilities getCapabilities(BearerToken userToken) throws WdsServiceException {
+    return executionWithRetryTemplate(
+        listenerResetRetryTemplate, () -> wdsClient.capabilitiesApi(userToken).capabilities());
+  }
+
+  public boolean useSearchByIdsFilter(BearerToken userToken) {
+    try {
+      Capabilities wdsAppInstanceCapabilities = getCapabilities(userToken);
+      Map<String, Object> additionalProperties =
+          wdsAppInstanceCapabilities.getAdditionalProperties();
+
+      return additionalProperties != null && additionalProperties.containsKey("search.filter.ids");
+    } catch (WdsServiceException e) {
+      logger.warn(
+          "Failed to get capabilities of WDS app instance. Error: %s".formatted(e.getMessage()), e);
+      return false;
+    }
+  }
+
+  public WdsRecordResponseDetails getRecords(
+      String recordType, List<String> recordIds, BearerToken userToken) {
+    if (useSearchByIdsFilter(userToken))
+      return getRecordsUsingSearchFilter(recordType, recordIds, userToken);
+    else return getRecordsSerially(recordType, recordIds, userToken);
+  }
```

Comment:
the capabilities api is a very lightweight implementation inside WDS, so it's no big deal … but conceivably you could check and cache the capabilities response for some amount of time. WDS's response will only change infrequently: when we perform app updates, and then only if something significant has changed in the app. I wouldn't block this PR on any kind of caching though.

Reply:
Yes we had discussed this briefly and I think we would want to cache this if its start affecting the performance or as part of future slice whichever happens first. But it is good to know that is a very lightweight implementation.